test/e2e/util.go

```diff
@@ -285,20 +285,25 @@ func waitResourcesDeleted(t *testing.T, f *framework.Framework, e *spec.EtcdClus
 }
 
 func waitBackupDeleted(f *framework.Framework, e *spec.EtcdCluster) error {
-	return retryutil.Retry(5*time.Second, 5, func() (done bool, err error) {
+	err := retryutil.Retry(5*time.Second, 5, func() (done bool, err error) {
 		rl, err := f.KubeClient.ReplicaSets(f.Namespace).List(k8sutil.ClusterListOpt(e.Name))
 		if err != nil {
 			return false, err
 		}
 		if len(rl.Items) > 0 {
 			return false, nil
 		}
-		// TODO: check backup pod deleted. There is a graceful deletion that takes too long.
-
-		if !e.Spec.Backup.CleanupBackupsOnClusterDelete {
-			return true, nil
-		}
-
+		// TODO: check backup pod deleted. There is a graceful deletion that takes too long
+		return true, nil
+	})
+	if err != nil {
+		return fmt.Errorf("failed to wait backup RS deleted: %v", err)
+	}
+	// We don't delete storage. For example, in restore test, we set up the storage to restore.
```

Comment:
i do not know what this comment means.

we should comment on why we can early return here, why we do not need to check the detection of storage.